File: 412_Fizz_Buzz/initialSolution.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
    public List<String> fizzBuzz(int n) {
        Map<Integer, String> fizzBuzzMap = new HashMap<>();
        fizzBuzzMap.put(3, "Fizz");
        fizzBuzzMap.put(5, "Buzz");
        List<String> answerList = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            String stringToReplaceIntegerWith = "";
            for (Map.Entry<Integer, String> entry : fizzBuzzMap.entrySet()) {
                if (i % entry.getKey() == 0) {
                    stringToReplaceIntegerWith += entry.getValue();
                }
            }
            if (stringToReplaceIntegerWith.equals("")) {
                stringToReplaceIntegerWith += String.valueOf(i);
            }
            answerList.add(String.valueOf(stringToReplaceIntegerWith));
        }
        return answerList;
    }
}

File: 412_Fizz_Buzz/initialSolution.go
package fizzbuzz

import (
	"strconv"
)

func fizzBuzz(n int) []string {
	fizzBuzzMap := []struct {
		key   int
		value string
	}{
		{3, "Fizz"},
		{5, "Buzz"},
	}

	var answerList []string
	for i := 1; i <= n; i++ {
		var result string
		for _, entry := range fizzBuzzMap {
			if i%entry.key == 0 {
				result += entry.value
			}
		}
		if result == "" {
			result = strconv.Itoa(i)
		}
		answerList = append(answerList, result)
	}

	return answerList
}
